### PR TITLE
Beaker improvements & HF formatting

### PR DESCRIPTION
**Pull request by hamishivi**
This PR turns the beaker script into a command-line configurable thing, making it easier for people to use it without actually reading the script and understanding what's happening. @dwadden may turn this into something more tightly integrated with beaker-py.

Additionally, I've added support for just applying the HF tokenizer in the eval chat formatting. This required a little refactoring to pass the tokenizer into the chat template. This allows us just to reuse the chat template given in an HF tokenizer (saving us time in implementing and testing any given chat format, hopefully).

### CODE REVIEW

**Comment by yizhongw** [comment]
This looks nice!

I guess the scripts in `scripts/eval` should still work since the refactored eval code can still accept a specified template function? Did you test them?

File: scripts/submit_eval_jobs.py [hunk 1]
```diff
@@ -1,25 +1,35 @@
...
     default_yaml = f.read()
 d1 = yaml.load(default_yaml, Loader=yaml.FullLoader)
 
-# cluster = "ai2/general-cirrascale"
-cluster = "ai2/allennlp-cirrascale"
-# cluster = "ai2/general-cirrascale-a100-80g-ib"
-# cluster = "ai2/prior-elanding"
-num_gpus = 1
+cluster = args.cluster
+num_gpus = args.num_gpus
```
Comment (yizhongw) [inline scripts/submit_eval_jobs.py:30]: Later, the num_gpus will be dynamically changed based on the model size. So, the argument here might be confusing. Shall we move the hard-coded GPUs into some post init functions for these arguments?

File: scripts/submit_eval_jobs.py [hunk 2]
```diff
@@ -39,110 +49,16 @@
     "alpaca_eval",
 ]
 
-# model to evaluate, each in the followng format: model name, their beaker id, checkpoint subfolder
+# format: model name, their beaker id, checkpoint subfolder, tuned or base.
+# or: name, path, None, tuned or base
 models = [
```
Comment (yizhongw) [inline scripts/submit_eval_jobs.py:54]: Can we move these models into a spreadsheet or somewhere to make them easily sharable?

File: scripts/submit_eval_jobs.py (hunk 1 above)
Comment (hamishivi) [reply scripts/submit_eval_jobs.py:30]: Good point. Maybe for now I'll leave it alone.

File: scripts/submit_eval_jobs.py (hunk 2 above)
Comment (hamishivi) [reply scripts/submit_eval_jobs.py:54]: Yup, this is happening: https://beaker.org/ws/ai2/adapt-models/datasets?sort=created:descending&results=false&committed=true
@dwadden is thinking about how to share this better too.

**Comment by hamishivi** [comment]
Yup, I've given this some quick tests.

**Comment by natolambert** [comment]
I'd say adding tests with all this chat template stuff would be great, maybe I can do this next week to get started here.

**Comment by hamishivi** [comment]
Tests would be good, maybe in a future PR :)

**Pull request merged**